LDK/resources/web/LDK/panel/TabbedReportPanel.js

```diff
@@ -44,6 +44,10 @@ Ext4.define('LDK.panel.TabbedReportPanel', {
     showFilterOptionsTitle: false,
     showReportsOption: false,
 
+    // Has at least one report been shown, meaning that we will always try to render the report whenever the user clicks
```

Comment:
https://www.labkey.org/ONPRC/Support%20Tickets/issues-details.view?issueId=40945

which was a followup from this issue:

https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=36231

Reply:
OK, so I agree the behavior Heidi/Raymond describe is a bug. I think another way to state the problem is that the TabbedReportPanel should save the state of a given tab once loaded. when the user toggles tabs, the intention is to keep that content (which would preserve a custom view). 

I think the actual bug is more like this: in possiblyUpdateActiveReport(), the intention of the code is to check relevant filters, and only update that tab is the existing filters dont match the filters used to previously load that tab. Your proposal with reportShown probably accomplishes that (i.e. updates the value of showReport); however, I propose this instead: 

Delete the reportShown stuff, and add the following possiblyUpdateActiveReport(), around line 644. This explicitly ignored showReport when testing if the existing tab is up-to-date.  a more perfect scheme would probably delegate some of that to the filterType; however, this is a smaller change:

        var reportTab = tab.items[0];
        if (reportTab.filters){
            for (var i in filters){
                //NOTE: the intent of this code is to test the filter conditions of this report (i.e. IDs, groups). showReport has nothing to do with this.
                if (i === 'showReport') {
                    return;
                }

My reasoning is that the intent of 'showReport' is really more about whether to auto-load the tab when loading the page. It has nothing to do with the filter-state of a given tab. There's clearly a lot of legacy and non-ideal code in the logic of rendering reports, but I'd argue this is a little more specific and direct to the actual problem.

Reply:
@labkey-jeckels one tangential thought. the original issue involved a user loading a tab, changing the view, and losing that state when switching back. There's a clear bug that should be fixed; however, I would bet that it's very common for users to want this UI to show a different starting view than the site-wide default.  Two thoughts:

1) There is no reason to have a single instance of the tabbed animal report panel on PRIMe or any EHR. Lots of user groups have their own folders on the site, and it would be very simple for IT to configure mini animal report instances tailored to a sub-user group. These alternate instances could have a subset of reports, and a given report could have a different view load by default (load by default on tab != default view). This would be low effort and maintenance, but probably be very useful.

2) It would probably be nicer if instead of blowing away the QWP and making a new one when filters change, if the TabbedReportPanel instead kept that QWP instance but changed filters on it. This would mean user-added changes, like toggling view, would be preserved. This is more effort than I'd personally put into this code, but if you or I ever remake something like this it might be worth considering.

Reply:
I like your suggestion on the change to possiblyUpdateActiveReport() and just pushed a commit that removes the reportShown variable

Good points on the more general thoughts on use cases, but beyond the scope of this merge.